selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/workspace/AbstractTestWorkspaceInjector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.core.workspace;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.String.format;
+
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.MembersInjector;
+import com.google.inject.Provider;
+import com.google.inject.name.Names;
+import java.lang.reflect.Field;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+
+/**
+ * Injector for custom annotation {@link InjectTestWorkspace}.
+ *
+ * @author Katerina Foniok
+ */
+abstract class AbstractTestWorkspaceInjector<T> implements MembersInjector<T> {
```

Comment:
Let's make this class `public` to simplify the usage.

Reply:
If I make this class public I won't be able to use abstract method. Or am I wrong?

Reply:
_public_ class doesn't restrict usage of _public_ abstract method.